src/script/script.cpp

```diff
@@ -338,12 +339,15 @@ bool GetScriptOp(CScriptBase::const_iterator& pc, CScriptBase::const_iterator en
     return true;
 }
 
-bool IsOpSuccess(const opcodetype& opcode)
+bool IsOpSuccess(const opcodetype& opcode, uint8_t leaf_version)
 {
-    return opcode == 80 || opcode == 98 || (opcode >= 126 && opcode <= 129) ||
+    if (leaf_version == TAPROOT_LEAF_TAPSCRIPT) {
+        return opcode == 80 || opcode == 98 || (opcode >= 126 && opcode <= 129) ||
            (opcode >= 131 && opcode <= 134) || (opcode >= 137 && opcode <= 138) ||
            (opcode >= 141 && opcode <= 142) || (opcode >= 149 && opcode <= 153) ||
            (opcode >= 187 && opcode <= 254);
+    }
+    return true;
```

Comment:
For unknown leaf_versions, do we want to return true by default?

Reply:
I don't see how this is better with `SigVersion`, perhaps I'm missing some pattern matching features in c++ that would enforce exhaustiveness checks?

Reply:
So when you say that I assume you want this 

```c++
bool IsOpSuccess(const opcodetype& opcode, SigVersion sigversion)
{
    if (sigversion == SigVersion::TAPSCRIPT) {
        return opcode == 80 || opcode == 98 || (opcode >= 126 && opcode <= 129) ||
           (opcode >= 131 && opcode <= 134) || (opcode >= 137 && opcode <= 138) ||
           (opcode >= 141 && opcode <= 142) || (opcode >= 149 && opcode <= 153) ||
           (opcode >= 187 && opcode <= 254);
    }
```

However this gives compiler warnings

```
script/script.cpp: In function ‘bool IsOpSuccess(const opcodetype&, SigVersion)’:
script/script.cpp:368:1: warning: control reaches end of non-void function [-Wreturn-type]
  368 | }
      | ^
script/script.cpp: In function ‘bool IsOpSuccess(const opcodetype&, SigVersion)’:
script/script.cpp:368:1: warning: control reaches end of non-void function [-Wreturn-type]
  368 | }
      | ^
```

The compiler warning makes sense to me - it wants a 'catch all' or 'default' case.